core/templates/pages/exploration-editor-page/services/exploration-automatic-text-to-speech.service.spec.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the ExplorationAutomaticTextToSpeechService.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+require('pages/exploration-editor-page/' +
+  'services/exploration-automatic-text-to-speech.service.ts');
+
+describe('Exploration Automatic Text To Speech Service', function() {
+  let eattss = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  beforeEach(angular.mock.inject(function($injector) {
+    eattss = $injector.get('ExplorationAutomaticTextToSpeechService');
+  }));
+
+  it('should test the child object properties', function() {
+    expect(eattss.propertyName).toBe('auto_tts_enabled');
+    expect(eattss._isValid(true)).toBe(true);
```

Comment:
Don't use private method here, find another way to test this method using any public method)

Reply:
@simonwex, is this comment resolved?

Reply:
Nice catch, @DubeySandeep

Reply:
@DubeySandeep you have any reference file so that I can replicate, though I have removed this and still getting 100% coverage

Reply:
I think you should be sure that you're testing the service behavior -- and then it will make 100% coverage. So take a look in which public methods these private methods are called and make assertions on it